lms/djangoapps/labster_accounts/management/commands/obfuscate_user_data.py

```diff
@@ -0,0 +1,154 @@
+"""
+Management command to obfuscate users data by email or in course.
+"""
+import string
+import random
+from itertools import chain
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import User
+
+from student.models import CourseEnrollment, CourseEnrollmentAllowed
+
+from labster_course_license.models import CourseLicense
+
+
+def action_decorator(action, field_name):
+    """
+    Decorator applies action to the provided value if value is not empty.
+    Prints logs for better process transparency.
+    """
+    def wrapper(val):
+        if val:
+            new_val = action(val)
+            print("Calling action `%s` for the attr `%s` with value `%s`. Resulting value `%s`" % (
+                action, field_name, val, new_val
+            ))
+            return new_val
+        else:
+            print("`%s` attr has empty value. Skip obfuscation" % field_name)
+            return val
+    return wrapper
+
+
+class Command(BaseCommand):
+    """
+    Obfuscate user credentials by course license or by email.
+    """
+    help = '''
+    Obfuscate user credentials for course license or by email.
+    '''
+
+    def add_arguments(self, parser):
+        """
+        Add arguments to the command parser.
+        """
+        parser.add_argument(
+            '--licenses',
+            dest='licenses',
+            nargs='+',
+            help='List of course licenses whose users should be obfuscated',
+        )
+        parser.add_argument(
+            '--emails',
+            dest='emails',
+            nargs='+',
+            help='List of user emails which should be obfuscated',
+        )
+
+    def handle(self, *args, **options):
+        """
+        Execute the command.
+        """
+        licenses = options['licenses']
+        emails = options['emails']
+
+        user_emails = set()
+
+        if licenses:
+            for lic in licenses:
+                print ("Processing license %s" % lic)
+                try:
+                    ccx = CourseLicense.objects.get(license_code=lic).course_id
+
+                    enrolled_users = CourseEnrollment.objects.filter(course_id=ccx).values_list(
+                        'user__email', flat=True
+                    ).exclude(user__email__contains='labster.com')
+                    not_enrolled_users = CourseEnrollmentAllowed.may_enroll_and_unenrolled(ccx).values_list(
+                        'email', flat=True
+                    )
+
+                    all_users = chain(not_enrolled_users, enrolled_users)
+
+                    user_emails |= set(all_users)
+                except CourseLicense.DoesNotExist:
+                    pass
```

Comment:
add a print that notifies us about this case.

Reply:
added